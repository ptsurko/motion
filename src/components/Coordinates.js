import React, { PureComponent } from 'react';
import Types from 'prop-types';

class Coordinates extends PureComponent {
  render() {
    const { participant } = this.props;

    return (
      <section className="phone-coords">
        <div>
          <span>X:&nbsp;</span>
          <span>{participant.beta}</span>
        </div>
        <div>
          <span>Y:&nbsp;</span>
          <span>{participant.gamma}</span>
        </div>
        <div>
          <span>Z:&nbsp;</span>
          <span>{participant.alpha}</span>
        </div>
      </section>
    );
  }
}

Coordinates.propTypes = {
  participant: Types.shape({
    alpha: Types.number.isRequired,
    beta: Types.number.isRequired,
    gamma: Types.number.isRequired,
  }).isRequired,
};

export default Coordinates;
