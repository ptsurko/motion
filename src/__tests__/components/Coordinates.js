import React from 'react';
import renderer from 'react-test-renderer';
import Coordinates from 'components/Coordinates';

it('renders Coordinates component correctly', () => {
  const participant = {
    alpha: 0,
    beta: 1,
    gamma: 2,
  };
  const tree = renderer.create(
    <Coordinates participant={participant} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
