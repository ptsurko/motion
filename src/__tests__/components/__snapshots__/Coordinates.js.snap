// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Coordinates component correctly 1`] = `
<section
  className="phone-coords"
>
  <div>
    <span>
      X: 
    </span>
    <span>
      1
    </span>
  </div>
  <div>
    <span>
      Y: 
    </span>
    <span>
      2
    </span>
  </div>
  <div>
    <span>
      Z: 
    </span>
    <span>
      0
    </span>
  </div>
</section>
`;
